src/Adapter/Product/CommandHandler/DeleteProductSupplierHandler.php

```diff
@@ -87,9 +87,7 @@ private function refreshProductDefaultSupplier(ProductSupplier $productSupplier)
         }
 
         $product->id_supplier = 0;
-        $product->supplier_reference = '';
```

Comment:
Oh ? I think previous code makes sense. Why delete ? Did I miss something ?

Reply:
because @PierreRambaud deprecated this field in some of his recent pr's

Reply:
Yes it was deprecated but until it is removed, we keep it functional.
See https://github.com/PrestaShop/PrestaShop/pull/20198/

We especially made sure to keep filling it with relevant data to preserve backward compatibility. Deprecating something does not mean "we break it" it means "it will be removed soon, but until it's removed we keep it working" (well ... in theory)

Reply:
Yes unfortunately we need to keep it working until the next major version.

Reply:
But what happens when a supplier is removed then? Should we reste the data as if no default supplier is assigned? Or should we pick one among the remaining ones? (In which case the reset will still be required when there's no more suppliers associated)

Reply:
I've reverted the changes. If default supplier is deleted then product->supplier_reference is reset to empty string and product->id_supplier is reset to 0.
Although I don't know if its ok. 
In case the default supplier is deleted, but there are still other suppliers then product will be left without a default supplier. Anyway, Im not sure if its worth to spend lots of time figuring this as it is  deprecated.